Source/Diagnostics/SliceDiagnostic.cpp

```diff
@@ -0,0 +1,490 @@
+                slice_lo[idim] = floor( ( (slice_cc_nd_box.lo(idim) 
+                                 - (real_box.lo(idim) + fac ) ) 
+                                 / dom_geom[0].CellSize(idim)) + fac * 1E-10);
+                slice_lo2[idim] = ceil( ( (slice_cc_nd_box.lo(idim) 
+                                 - (real_box.lo(idim) + fac) ) 
+                                 / dom_geom[0].CellSize(idim)) - fac * 1E-10 );    
+            }            
+            else {            
+                slice_lo[idim] =  round( (slice_cc_nd_box.lo(idim) 
+                                  - (real_box.lo(idim) ) ) 
+                                  / dom_geom[0].CellSize(idim));
+                slice_lo2[idim] = ceil((slice_cc_nd_box.lo(idim) 
+                                  - (real_box.lo(idim) ) ) 
+                                  / dom_geom[0].CellSize(idim) );
+            }
+ 
+            // flag for interpolation -- if reduced dimension location  //
+            //                           does not align with data point //
+            if ( slice_lo[idim] == slice_lo2[idim]) {            
+               if ( slice_cc_nd_box.lo(idim) - real_box.lo(idim) < fac ) {
+                  interp_lo[idim] = 1;
+               }
+            }
+            else {            
+               interp_lo[idim] = 1;
+            }
+
+            // ncells = 1 if dimension is reduced //
+            slice_hi[idim] = slice_lo[idim] + 1;
+        }
+        else
+        {
+            // moving realbox.lo and reabox.hi to nearest coarsenable grid point //
+            int index_lo = floor(((slice_realbox.lo(idim) - (real_box.lo(idim))) 
+                           / dom_geom[0].CellSize(idim) ) +fac * 1E-10);
+            int index_hi = ceil(((slice_realbox.hi(idim) - (real_box.lo(idim))) 
+                           / dom_geom[0].CellSize(idim) ) - fac * 1E-10);
+
+            bool modify_cr = true;
+
+            while ( modify_cr == true) {            
+                int lo_new = index_lo;
+                int hi_new = index_hi;
+                int mod_lo = index_lo % cr_ratio[idim];
+                int mod_hi = index_hi % cr_ratio[idim];
+                modify_cr = false;
+
+                // To ensure that the index.lo is coarsenable //
+                if ( mod_lo > 0) {
+                   lo_new = index_lo - mod_lo;
+                }
+                // To ensure that the index.hi is coarsenable //
+                if ( mod_hi > 0) {
+                   hi_new = index_hi + (cr_ratio[idim] - mod_hi);
+                }
+
+                // If modified index.hi is > baselinebox.hi, reduce cr ratio // 
+                // and provide more points that asked for                    //
+                if ( (hi_new * dom_geom[0].CellSize(idim)) 
+                      > real_box.hi(idim) - real_box.lo(idim) )
+                {
+                   cr_ratio[idim] = cr_ratio[idim]/2;
```

Comment:
Does this mean that the coarsening ratio used to dump the data may actually not be the one provided in `input`?

Reply:
Yes -- only under certain conditions. 
If the provided value of cr_ratio is not an exponent of 2. 
or if the coordinates provided is not coarsenable using the input values, I change the coordinates. 
But if the modified coordinates lie outside the domain, I modify the cr_ratio so that the new coordinates are coarsenable and lie within the simulation domain.